Makefile.toml

```
@@ -232,1396 +242,691 @@ fi
 '''
 ] }
 
-[tasks.setup]
+[tasks.install-twoliter]
 script = [
 '''
-# Ensure we use a supported architecture
-case "${BUILDSYS_ARCH}" in
-   x86_64|aarch64) ;;
-   *)
-      echo "Unrecognized architecture '${BUILDSYS_ARCH}'; please use 'x86_64 or 'aarch64'"
-      exit 1
-      ;;
-esac
-
-mkdir -p ${BUILDSYS_BUILD_DIR}
-mkdir -p ${BUILDSYS_OUTPUT_DIR}
-mkdir -p ${BUILDSYS_PACKAGES_DIR}
-mkdir -p ${BUILDSYS_STATE_DIR}
-mkdir -p ${GO_MOD_CACHE}
+cargo install \
+  --jobs ${BUILDSYS_JOBS} \
+  --locked \
+  --root "${TWOLITER_INSTALL_ROOT}" \
+  --quiet \
+  --git "${TWOLITER_GIT}" \
+  --rev "${TWOLITER_REV}" \
+  twoliter
 '''
 ]
 
+[tasks.setup]
+dependencies = ["install-twoliter"]
+command = "${TWOLITER}"
+args = [
+    "--log-level=${TWOLITER_LOG_LEVEL}",
+    "exec",
+    "setup",
+    "--project-path=${TWOLITER_PROJECT}",
+    "${@}",
+]
+
 [tasks.setup-build]
-dependencies = ["setup"]
-script = [
-'''
-for cmd in docker gzip lz4; do
-  if ! command -v ${cmd} >/dev/null 2>&1 ; then
-    echo "required program '${cmd}' not found" >&2
-    exit 1
-  fi
-done
-'''
+dependencies = ["install-twoliter"]
+command = "${TWOLITER}"
+args = [
+    "--log-level=${TWOLITER_LOG_LEVEL}",
+    "exec",
+    "setup-build",
+    "--project-path=${TWOLITER_PROJECT}",
+    "--cargo-home=${CARGO_HOME}",
+    "${@}",
 ]
 
 [tasks.fetch]
-dependencies = [
-  "fetch-sdk",
-  "fetch-toolchain",
-  "fetch-sources",
-  "fetch-vendored",
+dependencies = ["install-twoliter"]
+command = "${TWOLITER}"
+args = [
+    "--log-level=${TWOLITER_LOG_LEVEL}",
+    "exec",
+    "fetch",
+    "--project-path=${TWOLITER_PROJECT}",
+    "--cargo-home=${CARGO_HOME}",
```

nit: not sure if `cargo-make` supports this, but it might be possible to consolidate and standardize this as --
```suggestion
    "${TWOLITER_ARGS[*]}"
```